python/lsst/atmospec/utils.py

```diff
@@ -507,3 +512,107 @@ def getLinearStagePosition(exp):
+    -------
+    spectraction : `lsst.atmospec.spectraction.Spectraction`
+        The extracted spectraction object.
+
+    Notes
+    -----
+    Any ConfigurableInstances in supplied task config overrides will be
+    ignored. Currently (see DM-XXXXX) this causes a RecursionError.
+    """
+    # TODO move this to summit.utils to avoid the late import here which
+    # is required to avoid the cirucular import, and for this package to
+    # not depend on utils code.
+    import lsst.summit.utils.butlerUtils as butlerUtils
+
+    def makeQuery(dataId):
+        dayObs = butlerUtils.getDayObs(dataId)
+        seqNum = butlerUtils.getSeqNum(dataId)
+        queryString = (f"exposure.day_obs={dayObs} AND "
+                       f"exposure.seq_num={seqNum} AND "
+                       "instrument='LATISS'")
+
+        return queryString
+    repo = "LATISS" if not embargo else "/repo/embargo"
+
+    # TODO: use LATISS_DEFAULT_COLLECTIONS here?
+    butler = SimplePipelineExecutor.prep_butler(repo,
+                                                inputs=['LATISS/raw/all',
+                                                        'refcats',
+                                                        'LATISS/calib',
+                                                        ],
+                                                output=outputCollection)
+
+    pipeline = Pipeline.fromFile("${ATMOSPEC_DIR}/pipelines/processStar.yaml")
+
+    for taskName, configClass in taskConfigs.items():
+        for option, value in configClass.items():
+            # connections require special treatment
+            if isinstance(value, configClass.ConnectionsConfigClass):
+                for connectionOption, connectionValue in value.items():
+                    pipeline.addConfigOverride(taskName,
+                                               f'{option}.{connectionOption}',
+                                               connectionValue)
+            # ConfigurableInstance has to be done with .retarget()
+            elif not isinstance(value, pexConfig.ConfigurableInstance):
+                pipeline.addConfigOverride(taskName, option, value)
+
+    for taskName, configDict in configOptions.items():
+        for option, value in configDict.items():
+            # ConfigurableInstance has to be done with .retarget()
+            if not isinstance(value, pexConfig.ConfigurableInstance):
+                pipeline.addConfigOverride(taskName, option, value)
+
+    query = makeQuery(dataId)
+    executor = SimplePipelineExecutor.from_pipeline(pipeline,
+                                                    where=query,
+                                                    root=repo,
+                                                    butler=butler)
+
+    logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+    quanta = executor.run()
+
+    butler = butlerUtils.makeDefaultLatissButler(extraCollections=[outputCollection])
+    spectractionQuantum = quanta[3]
```

Comment:
Are these accessible by name?  That might be safer than ensuring it will always be this component of the output graph.

Reply:
Not exactly - it's just a raw `list`, but I could do `spectractionQuantum = [q for q in quanta if q.taskName=='lsst.atmospec.processStar.ProcessStarTask'][0]` but I don't know whether that's better or worse. It's certainly more robust, but it's also certainly not pretty! Thoughts?

Reply:
Maybe a comment explaining what you expect to get and which one you want?  That way there's some instructions in case this changes in the future.

Reply:
I don't mind the slightly gross list comp tbh, I'll change it to that, and also add a comment as to why I'm doing that.